wp-content/themes/csisnuclearnetwork/template-parts/block-post.php

```diff
@@ -11,19 +11,114 @@
+	$event_day = date_i18n('d', strtotime($legacy_event_date));
+	$event_month = date_i18n('M', strtotime($legacy_event_date));
+	$event_year = date_i18n('Y', strtotime($legacy_event_date));
+	
+	$event_date = '<span class="post-block__event-month text--caps">' . $event_month . '</span><span class="post-block__event-day">' . $event_day . '</span><span class="post-block__event-year text--short">' . $event_year . '</span>';
+}
+
+$event_full_date = date_i18n( strtotime("$event_start_date $event_start_time"));
+$yesterday = strtotime('now');
+
+if ( $event_full_date >= $yesterday ) {
+	$is_future_event = true;
+}
+
+if ( $event_start_time && $event_info['event_end_time'] ) {
+	$start_time = wp_date('g:i', strtotime($event_start_time));
+	$end_time = wp_date('g:i A T', strtotime($event_info['event_end_time']));
+	$event_time = $start_time . ' - ' . $end_time;
+} elseif ( $event_start_time ) {
+	$event_time = wp_date('g:i A T', strtotime($event_start_time));
+}
+
+
 ?>
-<article <?php post_class('post-block post-block--post'); ?> id="post-<?php the_ID(); ?>">	
-		<?php if ( has_post_thumbnail() ) : ?>
+<article <?php post_class( $classes ); ?> id="post-<?php the_ID(); ?>">	
+		<?php if ( has_post_thumbnail() && $post_type != 'events' && !$is_monthly_newsletter ) : ?>
 			<a href="<?php the_permalink(); ?>" class="post-block__img" title="<?php the_title_attribute(); ?>">
 				<?php the_post_thumbnail( 'large' ); ?>
 			</a>
-		<?php endif; ?>
+		<?php endif; 
 
+if ( $is_monthly_newsletter ) {
+	nuclearnetwork_display_subtypes();
+	the_title( '<h3 class="post-block__title text--bold"><a href="' . esc_url( get_permalink() ) . '">', '</a></h3>' );
+	nuclearnetwork_posted_on('M j, Y');
+} elseif ( $post_type === 'news' ) { ?>
+	<a href="<?php echo esc_url( get_permalink() ); ?>" class="post-block__news-link">
+		<?php
+		the_title( '<h3 class="post-block__title text--bold">', '</h3>' );
+		nuclearnetwork_posted_on('Y'); ?>
+	</a>
 	<?php
-		the_title( '<h3 class="post-block__title"><a href="' . esc_url( get_permalink() ) . '">', '</a></h3>' );
-		the_excerpt();			
+	} elseif ( $post_type === 'events' ) { ?>
+		<div class="post-block__event-content">
+			<?php
+			if ( $event_date ) { ?>
+			<div class="post-block__event-date"><?php echo $event_date; ?></div>
+			<?php
+			}
+			nuclearnetwork_display_subtypes();
+			the_title( '<h3 class="post-block__title text--bold"><a href="' . esc_url( get_permalink() ) . '">', '</a></h3>' );
+			the_excerpt(); 
+			if ( isset($event_info['event_registration_link']) && !empty($event_info['event_registration_link']) && $is_future_event && !$legacy_event_date ) { ?>
+			<a href="<?php echo esc_url( $event_info['event_registration_link'] ); ?>" class="post-block__register btn btn--outline-blue">Register<?php echo nuclearnetwork_get_svg( 'arrow-external' ); ?></a>
+			<?php
+			} ?>
+		</div>
+		<?php
+		if ( $event_start_date || $event_location ) { ?>
+		<div class="post-block__event-meta">
+			<?php
```

Comment:
I don't think you need to change these now, but in the future I would recommend making these functions in the `template-tags.php` file just like the functions to display the list of categories, tags, etc. It keeps your functional logic separated from your display logic, and makes it easier to call this information in multiple places if needed.